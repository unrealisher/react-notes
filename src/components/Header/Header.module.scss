.header {
  box-sizing: border-box;
  position: fixed;
  z-index: 10;
  display: flex;
  flex-direction: column;
  width: 100%;
  background-color: white;
}

.wrapper {
  background-color: white;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 13px 16px;
  z-index: 2;
}

.label {
  cursor: pointer;
  min-height: 16px;
}

.navigation_wrapper {
  position: absolute;
  width: 100%;
  top: -300%;
  transition: 0.6s;
  z-index: 1;
}

.checkbox:checked + .navigation_wrapper {
  top: 100%;
  transition: 0.6s;
  display: block;
}

@media (min-width: 970px) {
  .header {
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    padding-right: 16px;
  }

  .label {
    display: none;
  }

  .navigation_wrapper {
    position: static;
    width: auto;
  }
}
