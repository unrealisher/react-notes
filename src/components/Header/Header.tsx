import React from "react";

import Logo from "./../Logo/Logo";
import BurgerButton from "./../BurgerButton/BurgerButton";
import Navigation from "./../Navigation/Navigation";

import styles from "./Header.module.scss";

interface IProps {
  setPopup: Function;
}

const Header = (props: IProps): JSX.Element => {
  const { setPopup } = props;
  return (
    <div className={styles.header}>
      <h1 hidden>Яндекс Заметки</h1>
      <div className={styles.wrapper}>
        <Logo />
        <label className={styles.label} htmlFor="menu-checkbox">
          <BurgerButton />
        </label>
      </div>
      <input
        className={styles.checkbox}
        id="menu-checkbox"
        type="checkbox"
        hidden
      />
      <div className={styles.navigation_wrapper}>
        <Navigation setPopup={setPopup} />
      </div>
    </div>
  );
};

export default Header;
