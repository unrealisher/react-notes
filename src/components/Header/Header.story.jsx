import React from "react";
import { Provider } from "react-redux";
import store from "./../../store/index";

import Header from "./Header";

import { storiesOf } from "@storybook/react";

import "./../../index.scss";

storiesOf("Header", module).add("Десктоп", () => (
  <Provider store={store}>
    <Header />
  </Provider>
));
