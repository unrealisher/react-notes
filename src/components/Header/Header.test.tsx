import React from "react";
import { shallow } from "enzyme";
import { Header } from "./Header";

describe("Header", () => {
  const setPopup = (): void => {};
  it("correct render", () => {
    const component = shallow(<Header setPopup={setPopup} />);
    expect(component).toMatchSnapshot();
  });
});
