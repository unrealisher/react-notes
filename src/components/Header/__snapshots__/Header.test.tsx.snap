// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    setPopup={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          hidden={true}
        >
          Яндекс Заметки
        </h1>,
        <div
          className="wrapper"
        >
          <Logo />
          <label
            className="label"
            htmlFor="menu-checkbox"
          >
            <BurgerButton />
          </label>
        </div>,
        <input
          className="checkbox"
          hidden={true}
          id="menu-checkbox"
          type="checkbox"
        />,
        <div
          className="navigation_wrapper"
        >
          <Connect(Navigation)
            setPopup={[Function]}
          />
        </div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Яндекс Заметки",
          "hidden": true,
        },
        "ref": null,
        "rendered": "Яндекс Заметки",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Logo />,
            <label
              className="label"
              htmlFor="menu-checkbox"
            >
              <BurgerButton />
            </label>,
          ],
          "className": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <BurgerButton />,
              "className": "label",
              "htmlFor": "menu-checkbox",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "label",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "checkbox",
          "hidden": true,
          "id": "menu-checkbox",
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Navigation)
            setPopup={[Function]}
          />,
          "className": "navigation_wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "setPopup": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            hidden={true}
          >
            Яндекс Заметки
          </h1>,
          <div
            className="wrapper"
          >
            <Logo />
            <label
              className="label"
              htmlFor="menu-checkbox"
            >
              <BurgerButton />
            </label>
          </div>,
          <input
            className="checkbox"
            hidden={true}
            id="menu-checkbox"
            type="checkbox"
          />,
          <div
            className="navigation_wrapper"
          >
            <Connect(Navigation)
              setPopup={[Function]}
            />
          </div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Яндекс Заметки",
            "hidden": true,
          },
          "ref": null,
          "rendered": "Яндекс Заметки",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Logo />,
              <label
                className="label"
                htmlFor="menu-checkbox"
              >
                <BurgerButton />
              </label>,
            ],
            "className": "wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <BurgerButton />,
                "className": "label",
                "htmlFor": "menu-checkbox",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "label",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "checkbox",
            "hidden": true,
            "id": "menu-checkbox",
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Navigation)
              setPopup={[Function]}
            />,
            "className": "navigation_wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "setPopup": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
